mono/mini/debugger-agent.c

```diff
@@ -3331,6 +3339,21 @@ compute_frame_info (MonoInternalThread *thread, DebuggerTlsData *tls, gboolean f
 	tls->frames = new_frames;
 	tls->frame_count = new_frame_count;
 	tls->frames_up_to_date = TRUE;
+
+	if (CHECK_PROTOCOL_VERSION (2, 52)) {
+		MonoJitTlsData *jit_data = thread->thread_info->jit_data;
+		gboolean has_interp_resume_state = FALSE;
+		MonoInterpFrameHandle interp_resume_frame = NULL;
+		gpointer interp_resume_ip = 0;
+		mini_get_interp_callbacks ()->get_resume_state (jit_data, &has_interp_resume_state, &interp_resume_frame, &interp_resume_ip);
+		if (has_interp_resume_state && tls->frame_count > 0) {
+			StackFrame *top_frame = tls->frames [0];
+			if (interp_resume_frame == top_frame->interp_frame) {
+				int native_offset = (int) ((uintptr_t) interp_resume_ip - (uintptr_t) top_frame->de.ji->code_start);
```

Comment:
why does `top_frame->native_offset` not have the correct value here?

Reply:
`top_frame->de.native_offset` does not have the correct value here because it's populated by `interp_frame_iter_next`, which does not retrieve data from the interpreter's resume state (if present and valid). An alternative would instead entail altering `interp_frame_iter_next`, so that it does retrieve data from the interpreter's resume state, although then (if https://github.com/mono/mono/pull/14667#issuecomment-496553352 is important), I'd need a way to thread a "should read from the resume state" flag through.